rwengine/include/engine/GameInputState.hpp

```diff
@@ -0,0 +1,58 @@
+#ifndef RWENGINE_GAMEINPUTSTATE_HPP
+#define RWENGINE_GAMEINPUTSTATE_HPP
+
+struct GameInputState
+{
+	/// Inputs that can be controlled
+	/// @todo find any sensible values
+	enum Control {
+		/* On Foot */
+		FireWeapon = 0,
+		NextWeapon,  // Also Next Target
+		LastWeapon,  // Also Last Target
+		GoForward,
+		GoBackwards,
+		GoLeft,
+		GoRight,
+		ZoomIn,
+		ZoomOut,
+		EnterExitVehicle,
+		ChangeCamera,
+		Jump,
+		Sprint,
+		AimWeapon,
+		Crouch,
+		Walk,
+		LookBehind,
+
+		/* In Vehicle */
+		VehicleFireWeapon,
```

Comment:
The buttons for firing while in a vehicle are the same as on foot, so  `VehicleFireWeapon` in unneeded. Some of the directions are also the same. I'll remove the redundant ones.